wagtail/admin/widgets/chooser.py

```diff
@@ -277,9 +277,12 @@ def base_js_init_options(self):
         }
 
     def get_instance(self, value):
-        instance = super().get_instance(value)
-        if instance:
-            return instance.specific
+        try:
+            instance = super().get_instance(value)
+            if instance:
+                return instance.specific
+        except Page.DoesNotExist:
```

Comment:
Just wondering if this is really the right place to patch this...

If we want choosers to silently drop anything that isn't a recognised ID for the model being chosen (which is probably a sensible behaviour - it's presumably how a plain dropdown-based model chooser widget would behave) then it would make sense for this to go in the `BaseChooser.get_instance` method. On the other hand, if we just want to handle the specific case of the move page form (where we know of a legitimate reason for the ID to be invalid), it would be best to validate it in that view/form before it reaches the chooser widget.

Reply:
I think it may be best to try to handle it in BaseChooser. Will update this accordingly

Reply:
Updated